core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -401,8 +415,9 @@
 
   EvictBufferTest
   (evict-cached-buffer! [_ k]
-    (doto (.remove memory-store k)
-      (util/close)))
+    (when-let [buf (.getIfPresent memory-store k)]
+      (.invalidate memory-store k)
+      (util/close buf)))
```

Comment:
might this be a double-close? I'm assuming the invalidation will cause a close too

Reply:
I think the reason I added it (although not consciously), is that you run into the same issue as with `invalidateAll` above (i.e. the `removalListener` running somehow asynchronously).

Reply:
See above.